nitransforms/cli.py

```diff
@@ -0,0 +1,122 @@
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from textwrap import dedent
+import sys
+
+from .linear import load as linload
+from .nonlinear import load as nlinload
+
+
+def cli_apply(pargs):
+    """
+    Apply a transformation to an image, resampling on the reference
+
+    Sample usage:
+
+        $ nt apply xform.fsl moving.nii.gz --ref reference.nii.gz
+
+        $ nt apply warp.nii.gz moving.nii.gz --fmt afni --nonlinear
+
+    """
+    fmt = pargs.fmt or pargs.transform.split('.')[-1]
+    if fmt == 'tfm':
+        fmt = 'itk'
+    elif fmt == 'lta':
+        fmt = 'fs'
+
+    if fmt not in ('fs', 'itk', 'fsl', 'afni'):
+        raise RuntimeError(
+            "Cannot determine transformation format, manually set format with the `--fmt` flag"
+        )
+
+    if pargs.nonlinear:
+        xfm = nlinload(pargs.transform, fmt=fmt, reference=pargs.ref)
+    else:
+        xfm = linload(pargs.transform, fmt=fmt)
+
+    xfm.apply(
+        pargs.moving,
+        order=pargs.order,
+        mode=pargs.mode,
+        cval=pargs.cval,
+        prefilter=pargs.prefilter
+    )
+
+
+def get_parser():
+    desc = dedent("""
```

Comment:
Does not argparse compose a good description for commands?

Reply:
i find it's nice to have a little blurb describing the subparsers at the top level (argparse doesn't do this)